.github/workflows/nessie.yaml

```diff
@@ -166,11 +182,34 @@ jobs:
         run: |
             cd nessie/ops
             docker-compose ps -q postgres && docker-compose exec -T postgres pg_dumpall --username=lakefs | gzip | aws s3 cp - s3://nessie-system-testing/${{ github.run_number }}/dump.gz
+
+      - name: Test lakeFS S3 with Spark
+        env:
+          TAG: ${{ steps.version.outputs.tag }}
+          LAKEFS_STATS_ENABLED: "false"
+          LAKEFS_BLOCKSTORE_TYPE: s3
+          LAKEFS_GATEWAYS_S3_DOMAIN_NAME: s3.local.lakefs.io:8000
+          DOCKER_REG: ${{ steps.login-ecr.outputs.registry }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.NESSIE_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.NESSIE_AWS_SECRET_ACCESS_KEY }}
+          NESSIE_TEST_DATA_ACCESS: true,false
+          NESSIE_STORAGE_NAMESPACE: s3://nessie-system-testing/${{ github.run_number }}
+          NESSIE_AWS_ACCESS_KEY_ID: ${{ secrets.NESSIE_AWS_ACCESS_KEY_ID }}
+          NESSIE_AWS_SECRET_ACCESS_KEY: ${{ secrets.NESSIE_AWS_SECRET_ACCESS_KEY }}
+          TESTER_ACCESS_KEY_ID: ${{ secrets.TESTER_ACCESS_KEY_ID }}
+          TESTER_SECRET_ACCESS_KEY: ${{ secrets.TESTER_SECRET_ACCESS_KEY }}
+        run: |
+            docker-compose -f nessie/ops/docker-compose.yaml down -v || true
+            docker-compose -f nessie/ops/docker-compose.yaml up -d
+            docker-compose -f nessie/ops/docker-compose.yaml exec -T lakefs /app/wait-for localhost:8000
+            docker-compose -f nessie/ops/docker-compose.yaml exec -T lakefs lakefs setup --user-name tester --access-key-id ${TESTER_ACCESS_KEY_ID} --secret-access-key ${TESTER_SECRET_ACCESS_KEY}
+            docker-compose -f nessie/ops/docker-compose.yaml exec -T lakefs lakectl --config /lakefs/nessie/ops/lakectl-tester.yaml repo create lakefs://example ${NESSIE_STORAGE_NAMESPACE} -d master
+            docker-compose -f nessie/ops/docker-compose.yaml exec -T lakefs lakectl --config /lakefs/nessie/ops/lakectl-tester.yaml fs upload -s /lakefs/nessie/spark-app/data-sets/sonnets.txt lakefs://example/master/sonnets.txt
+            docker-compose -f nessie/ops/docker-compose.yaml run -T --no-deps --rm spark-submit spark-submit --master spark://spark:7077 -c "spark.hadoop.fs.s3a.access.key=${TESTER_ACCESS_KEY_ID}" -c "spark.hadoop.fs.s3a.secret.key=${TESTER_SECRET_ACCESS_KEY}" --class Sonnets /lakefs/nessie/spark-app/target/scala-2.12/sonnets_2.12-0.1.jar
```

Comment:
AFAIU this tests that the Spark run succeeded (because otherwise `spark-submit` fails).  I would prefer to validate a bit more -- e.g. the app or another step here should verify that an output file was created, so we also test Spark writers.